pkg/newlog/cmd/newlogd.go

```diff
@@ -1478,13 +1475,32 @@ func findMovePrevLogFiles(movefile chan fileChanInfo) {
 
 	// on prev life's dev-log and app-log
 	for _, f := range files {
+		if f.IsDir() {
+			continue
+		}
+
 		isDev := strings.HasPrefix(f.Name(), devPrefix)
 		isApp := strings.HasPrefix(f.Name(), appPrefix)
-		if !f.IsDir() && (isDev && len(f.Name()) > len(devPrefix) || isApp && len(f.Name()) > len(appPrefix)) {
-			var fileinfo fileChanInfo
-			prevLogFile := collectDir + "/" + f.Name()
-			fileinfo.tmpfile = prevLogFile
-			fileinfo.isApp = isApp
+
+		if (isDev && len(f.Name()) > len(devPrefix)) || (isApp && len(f.Name()) > len(appPrefix)) {
+			fileinfo := fileChanInfo{
+				tmpfile: path.Join(collectDir, f.Name()),
+				isApp:   isApp,
+			}
+			if isDev {
+				fileinfo.notUpload = strings.HasPrefix(f.Name(), devPrefixKeep)
+			} else {
+				// this is going to be executed right after bootup, so the availability of config for this app is subject to race condition
+				// furthermore the config might not contain the appUUID anymore, so we are better off uploading the logs as default
+				appuuid := getAppuuidFromLogfile(fileinfo)
+				if val, found := domainUUID.Load(appuuid); found {
```

Comment:
Is it available after the reboot? Or it's not a problem if it is not?

Reply:
That depends if the domainconfig is available through pubsub already. For this zedagent needs to get device config, parse it, inform zedmanager, which needs to inform domainmanager. So I guess it's unlikely all this happens before newlogd starts :)
That's why we also set a default.

Reply:
So, there is no race between "stored" and a default value?

Reply:
the race is between how fast newlogd gets to this piece of code vs how quickly domainConfig becomes available through pubsub

Reply:
I mean, could it be a problem if we set the default value and ignore the real value (because of the race)? Am I right that even if it happens, we still wouldn't lose any logs that haven't been uploaded?